addon/mixins/adapter-fetch.js

```diff
@@ -211,44 +221,54 @@ export default Ember.Mixin.create({
   },
 
   /**
+   * @param {Object} adapter
    * @param {Object} response
-   * @param {Promise} bodyPromise
+   * @param {Object} payload
    * @param {Object} requestData
    * @override
    */
-  ajaxSuccess(response, bodyPromise, requestData) {
-    const headersObject = headersToObject(response.headers);
+  ajaxSuccess(adapter, response, payload, requestData) {
+    const returnResponse = adapter.handleResponse(
+      response.status,
+      headersToObject(response.headers),
+      payload,
+      requestData
+    );
 
-    return bodyPromise.then((body) => {
-      const returnResponse = this.handleResponse(
-        response.status,
-        headersObject,
-        body,
-        requestData
-      );
+    if (returnResponse && returnResponse.isAdapterError) {
+      return RSVP.Promise.reject(returnResponse);
+    } else {
+      return returnResponse;
+    }
+  },
 
-      if (returnResponse && returnResponse.isAdapterError) {
-        return RSVP.Promise.reject(returnResponse);
-      } else {
-        return returnResponse;
-      }
-    });
+
+/**
+ * Allows for the error to be selected from either the
+ * response object, or the response data.
+ * @param {Object} response
+ * @param {Object} responseData
+ */
+  parseFetchResponseForError(response, responseData) {
+    return responseData;
   },
 
   /**
-   * @param {Error} error
+   * @param {Object} adapter
    * @param {Object} response
    * @param {Object} requestData
+   * @param {Object} responseData
```

Comment:
ya, I guess this changes a bit from ember data's method, https://github.com/emberjs/data/blob/171102c0106917fd4eae206f5e86e65b716d7713/addon/adapters/rest.js#L1507 where they are able to pull the response out of the `jqXHR` while due to how the response works with fetch, we are unable to do that directly. I'll think about this and get back to you.